internal/resource-request-operator/resourceRequest_controller.go

```diff
@@ -0,0 +1,189 @@
+package resourceRequestOperator
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	discoveryv1alpha1 "github.com/liqotech/liqo/apis/discovery/v1alpha1"
+	sharingv1alpha1 "github.com/liqotech/liqo/apis/sharing/v1alpha1"
+	utils "github.com/liqotech/liqo/pkg/liqonet"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+// ResourceRequestReconciler reconciles a ResourceRequest object
+type ResourceRequestReconciler struct {
+	client.Client
+	Log       logr.Logger
+	Scheme    *runtime.Scheme
+	ClusterId string
+}
+
+const (
+	finalizerName = "discovery.liqo.io.resourceRequest.finalizer"
+	offerPrefix   = "resourceoffer-"
+)
+
+// +kubebuilder:rbac:groups=discovery.liqo.io,resources=resourceRequests,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=discovery.liqo.io,resources=resourceRequests/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=sharing.liqo.io,resources=resourceOffers,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=sharing.liqo.io,resources=resourceOffers/status,verbs=get;update;patch
+
+func (r *ResourceRequestReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+
+	var resourceRequest discoveryv1alpha1.ResourceRequest
+	err := r.Get(ctx, req.NamespacedName, &resourceRequest)
+	if err != nil && !apierrors.IsNotFound(err) {
+		klog.Errorf("ResourceRequestReconciler-> unable to get resource %s: %s", req.NamespacedName, err)
+		return ctrl.Result{}, err
+	}
+
+	if apierrors.IsNotFound(err) {
+		return ctrl.Result{}, nil
+	}
+
+	//not in delete
+	if resourceRequest.DeletionTimestamp.IsZero() {
+		// add finalizer
+		if !utils.ContainsString(resourceRequest.ObjectMeta.Finalizers, finalizerName) {
+			resourceRequest.Finalizers = append(resourceRequest.Finalizers, finalizerName)
+			if err := r.Update(ctx, &resourceRequest); err != nil {
+				klog.Errorf("%s -> unable to update resource %s: %s", resourceRequest.Spec.ClusterIdentity, resourceRequest.Spec.Namespace, err)
+				return ctrl.Result{}, err
+			}
+
+		}
+	} else {
+		//the resource is being deleted
+		if utils.ContainsString(resourceRequest.ObjectMeta.Finalizers, finalizerName) {
+			if err := r.deleteOffer(); err != nil {
+				klog.Errorf("Error deleting offer")
+				return ctrl.Result{}, err
+			}
```

Comment:
I would add a few comments here explaining why the offer is deleted. In particular, can we use the garbage collection based on ownerReference instead of an explicit delete?